recipe-server/client/control_new/components/recipes/Listing.js

```diff
@@ -0,0 +1,185 @@
+import { Button, Col, Input, Pagination, Row, Table } from 'antd';
+import autobind from 'autobind-decorator';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { push as pushAction, Link } from 'redux-little-router';
+
+import BooleanIcon from 'control_new/components/common/BooleanIcon';
+import LoadingOverlay from 'control_new/components/common/LoadingOverlay';
+import QueryRecipes from 'control_new/components/data/QueryRecipes';
+import ListingActionBar from 'control_new/components/recipes/ListingActionBar';
+import DataList from 'control_new/components/tables/DataList';
+import * as recipeActions from 'control_new/state/recipes/actions';
+import {
+  getRecipeListingColumns,
+  getRecipeListingCount,
+  getRecipeListingFlattenedAction,
+} from 'control_new/state/recipes/selectors';
+import {
+  getCurrentURL,
+  getQueryParameter,
+} from 'control_new/state/router/selectors';
+
+
+@connect(
+  state => ({
+    columns: getRecipeListingColumns(state),
+    count: getRecipeListingCount(state),
+    getCurrentURL: queryParams => getCurrentURL(state, queryParams),
+    ordering: getQueryParameter(state, 'ordering', '-last_updated'),
+    pageNumber: parseInt(getQueryParameter(state, 'page', 1), 10),
+    recipes: getRecipeListingFlattenedAction(state),
+    searchText: getQueryParameter(state, 'searchText'),
+    status: getQueryParameter(state, 'status'),
+  }),
+  dispatch => (
+    bindActionCreators({
+      fetchFilteredRecipesPage: recipeActions.fetchFilteredRecipesPage,
+      push: pushAction,
+    }, dispatch)
+  )
+)
+@autobind
+export default class Listing extends React.Component {
+  static propTypes = {
+    columns: PropTypes.object,
+    count: PropTypes.number,
+    fetchFilteredRecipesPage: PropTypes.func.isRequired,
```

Comment:
I understand this is alphabetized, but I prefer to keep the `isRequired`s towards the top of the propTypes, so it's easier to grok when looking over a component.

Perhaps we could hoist requireds to the top, then alphabetize non/requireds separately, e.g.:

```
...
fetchFilteredRecipesPage: PropTypes.func.isRequired,
getCurrentURL: PropTypes.func.isRequired,
columns: PropTypes.object,
count: PropTypes.number,
...
```